app/services/database.py

```diff
@@ -16,28 +21,45 @@ def __init__(self, table_name=None):
 
     def load_data(self, table_name=None):
 
-        if table_name is None or len(table_name) is 0:
-            table_name = DATA_PATH
+        gcp_data = gcp_bucket.read_table(table_name)
+        if gcp_data:
+            return gcp_data.get("tmp_uas")
+        else:
+            try:
+                if not table_name:
+                    table_name = DATA_DUMP_FILE_NAME
+
+                with open(table_name, "r") as f:
+                    items = json.load(f).get("tmp_uas")
+                    if len(items) > 0:
+                        logger.info(f"Call returned {len(items)} items after reading local file.")
+                    else:
+                        logger.warning(f"Call returned {len(items)} items after reading local file.")
 
-        with open(table_name, "r") as f:
-            return json.load(f)["tmp_uas"]
+                    return json.load(f).get("tmp_uas")
+            except Exception:
+                logging.exception("Could not find file locally and something went wrong when looking for file"
+                                  "in GCP.")
+                raise ValueError("Could not find file locally and something went wrong when looking for file in GCP.")
 
-    def filter_users_with_inclusion_list(self, inclusion_list, limit, db_table=None):
+    def filter_users_with_inclusion_list(self, inclusion_list, user_limit, db_table=None):
 
         if db_table is None:
             db_table = self.table
 
         if len(inclusion_list) == 0:
             raw_inclusion_list = get_unique_vals_for_property(db_table, "anon_id")
-            raw_inclusion_list = raw_inclusion_list[:int(limit)]
+            raw_inclusion_list = raw_inclusion_list[:int(user_limit)]
         else:
-            inclusion_list = inclusion_list[:int(limit)]
+            inclusion_list = inclusion_list[:int(user_limit)]
             raw_inclusion_list = [user.id for user in inclusion_list]
 
         data_with_inclusion_filtered = [row for row in db_table[:] if row["anon_id"] in raw_inclusion_list]
-
-        if len(data_with_inclusion_filtered) == 0:
-            print("No items returned as user/users specified are not available.")
+        if len(data_with_inclusion_filtered) > 0:
```

Comment:
hmm looking at this I think you were right about the logging, it did look tidier with the function. I'm wondering if we could keep just the logger.warning('') message when no items are returned.

Reply:
Hmmm I don't think it should be a WARNING when >0 items are returned and I think it should be a warning when zero items are returned. Zero returned items is pretty helpful for debugging. I guess we could make all the logs .INFO. That is the most simple solution.

Reply:
Oh I think i follow what you mean now - only log if no items are returned. I think it's quite useful to see the number of returned items in the logs.

Reply:
Final version logs everything as .INFO. After discussion with Tati we decided that we could create a stackdriver metrics  filter that would flag a warning when zero items are returned.